lib/profile.js

```diff
@@ -1,24 +1,55 @@
 'use strict';
 
-function configure (collection, storage) {
+var ObjectID = require('mongodb').ObjectID;
 
+function storage (collection, storage) {
+	
   function create (obj, fn) {
     obj.created_at = (new Date( )).toISOString( );
-    api( ).insert(obj, function (err, doc) {
+    api().insert(obj, function (err, doc) {
       fn(null, doc);
     });
   }
 
+  function save (obj, fn) {
+	obj._id = new ObjectID(obj._id);
+    obj.created_at = (new Date( )).toISOString( );
+    api().save(obj, function (err, doc) {
+      fn(err, doc);
+    });
+  }
+
   function list (fn) {
-    return api( ).find({ }).sort({created_at: -1}).toArray(fn);
+    return api().find({ }).sort({validfrom: -1}).toArray(fn);
```

Comment:
I don't think we should be introducing `validfrom` here, there is no data in that field yet to the sort will be undefined